packages/builders/src/components/button/mixins/EmojiOrLabelButtonMixin.ts

```diff
@@ -0,0 +1,44 @@
+import type { APIButtonComponent, APIButtonComponentWithSKUId, APIMessageComponentEmoji } from 'discord-api-types/v10';
+
+export interface EmojiOrLabelButtonData
+	extends Pick<Exclude<APIButtonComponent, APIButtonComponentWithSKUId>, 'emoji' | 'label'> {}
+
+export class EmojiOrLabelButtonMixin {
+	protected declare readonly data: EmojiOrLabelButtonData;
+
+	/**
+	 * Sets the emoji to display on this button.
+	 *
+	 * @param emoji - The emoji to use
+	 */
+	public setEmoji(emoji: APIMessageComponentEmoji) {
+		this.data.emoji = emoji;
+		return this;
+	}
+
+	/**
+	 * Clears the emoji on this button.
+	 */
+	public clearEmoji() {
+		this.data.emoji = undefined;
```

Comment:
Why not to [`delete`](<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete>) the property. Setting it to `undefined` will waste the body character limit for the request

Reply:
Delete de-optimizes objects and should always be avoided. We can work around the character limit in the appropriate place.

Reply:
> Why not to [delete](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete) the property?

It's much slower than setting it to `undefined`

> Setting it to `undefined´ will waste the body character limit for the request

It will not, since `JSON.stringify()` ignores keys with value `undefined`, so they will not be sent to Discord at all